rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c

```diff
@@ -225,13 +231,26 @@ static bool encl_build(struct sgx_secs *secs, void *bin, unsigned long bin_size,
 	if (!encl_create(dev_fd, bin_size, secs))
 		goto out_dev_fd;
 
+	uint64_t* ssa_frame = malloc(4096 * secs->ssa_frame_size);
+	if (ssa_frame == NULL){
+		fprintf(stderr, "Failed malloc memory for ssa_frame\n");
+		return false;
+	}
+	memset(ssa_frame, 0, 4096 * secs->ssa_frame_size);
+	uint64_t * addr = (uint64_t *)(bin + 16);
+	memcpy(addr, &bin_size, 8);
```

Comment:
You need to define tcs structure instead of this way. Refer to sgx_tcs defined at https://github.com/jsakkine-intel/linux-sgx/blob/v33/arch/x86/kernel/cpu/sgx/arch.h.